eng/generate-compiler-code.cs

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env dotnet
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// Need this fix to delete the static graph disable: https://github.com/dotnet/sdk/pull/50532
+#:property RestoreUseStaticGraphEvaluation=false
+#:project ../src/Tools/Source/CompilerGeneratorTools/Source/BoundTreeGenerator/
+#:project ../src/Tools/Source/CompilerGeneratorTools/Source/IOperationGenerator/
+#:project ../src/Tools/Source/CompilerGeneratorTools/Source/CSharpSyntaxGenerator/
+#:project ../src/Tools/Source/CompilerGeneratorTools/Source/CSharpErrorFactsGenerator/
+#:project ../src/Tools/Source/CompilerGeneratorTools/Source/VisualBasicSyntaxGenerator/
+#:project ../src/Tools/Source/CompilerGeneratorTools/Source/VisualBasicErrorFactsGenerator/
+
+using System.Runtime.CompilerServices;
+
+var repoRoot = GetRoslynDirectory();
+
+var test = false;
+var configuration = "Debug";
+
+for (var i = 0; i < args.Length; i++)
+{
+    var a = args[i];
+    switch (a)
+    {
+        case "-test":
+            test = true;
+            break;
+        case "-configuration" when i + 1 < args.Length:
+            configuration = args[i++];
+            break;
+        default:
+            Console.WriteLine("Unknown argument: " + a);
+            return 1;
+    }
+}
+
+var csharpDir = Path.Join(repoRoot, "src", "Compilers", "CSharp", "Portable");
+var vbDir = Path.Join(repoRoot, "src", "Compilers", "VisualBasic", "Portable");
+
+// Folders for test artifacts when running in -test mode
+var generationTempDirRoot = Path.Join(repoRoot, "artifacts", "log", configuration, "Generated");
+
+var generatedCSharpDir = Path.Join(csharpDir, "Generated");
```

Comment:
Consider breaking down the top-level logic into helpers.
The old implementation had a clear structure:
```
  Run-Language "CSharp" "cs" $csharpDir $csharpTestDir $csharpSyntaxProject $csharpErrorFactsProject
  Run-Language "VB" "vb" $basicDir $basicTestDir $basicSyntaxProject $basicErrorFactsProject 
  Run-IOperation $coreDir $operationsProject
  Run-GetText
```
which is less clear now.
I'd consider similarly extracting the logic to parse command-line into a helper as well.
Then the top-level steps are salient: parse, generate C#, generate VB, generate IOperation, and test

Reply:
Refactored for readability, let me know what you think